### ISSUE TITLE
Sigmoid fails and output all zeros

### ISSUE THREAD

**Issue by ZhangTianrong** [microsoft/onnxruntime#10154]
**Describe the bug**
A clear and concise description of what the bug is. To avoid repetition please make sure this is not one of the known issues mentioned on the respective release page.

I am trying to export a pytorch model to onnx. The last operation is sigmoid but onnxruntime's result is all zeros. I looked into the input to sigmoid too, and it's correct.

![image](https://user-images.githubusercontent.com/20651728/147724838-78615f89-427d-46d1-a682-c605156ece79.png)

**Urgency**
If there are particular important use cases blocked by this or strict project-related timelines, please share more information and dates. If there are no hard deadlines, please specify none.

none

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux-5.10.60.1-microsoft-standard-WSL2-x86_64-with-glibc2.31
- ONNX Runtime installed from (source or binary): pypi
- ONNX Runtime version: onnxruntime==1.10.0
- Python version: Python 3.9.7
- Visual Studio version (if applicable): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: 11.6
- GPU model and memory: NVIDIA GeForce RTX 3050 Ti Laptop GPU w/ 4G memory

**To Reproduce**
- Describe steps/code to reproduce the behavior.
- Attach the ONNX model to the issue (where applicable) to expedite investigation.

I just used the following code to export the model:

```python
torch.onnx.export(
                    self.net, 
                    inp, 
                    "det.onnx", 
                    export_params=True, 
                    opset_version=11, 
                    verbose=True,
                    do_constant_folding=True, 
                    input_names = ['input'], 
                    output_names = ['output'],
                    dynamic_axes={
                        'input' : {
                            0 : 'batch_size',
                            2 : 'height_in',
                            3 : 'width_in'
                        }
                    }
                )
```

Since the result before sigmoid is correct and the graph also look correct, there might be something wrong with sigmoid implementation I guess. I also tried some trivial model with only a sigmoid operation there and there isn't such an issue. Maybe this is related to the input as well?

[Link to model](https://1drv.ms/u/s!Ah7B82t6Uj6F5you7S2UXYdr9mCY?e=5X01Ik)
[Link to input data](https://1drv.ms/u/s!Ah7B82t6Uj6F5ysJxQi9Gwinz_d2?e=hPKxOy)

**Expected behavior**

I expect it to compute the correct results.

**Screenshots**

**Additional context**
Add any other context about the problem here. If the issue is about a particular model, please share the model details as well to facilitate debugging.

This is the [PaddleOCR](https://github.com/PaddlePaddle/PaddleOCR) text detection model (mobile version) for Chinese and English, which is quite small. I am using the pytorch version of it [here](https://github.com/frotms/PaddleOCR2Pytorch).

**Comment by yuslepukhin** [comment]
It would be nice to see the script that you are using to invoke ONNX Runtime. The onnxruntime package seems to be a CPU package. Is this correct?

**Comment by ZhangTianrong** [comment]
Yes, it's the CPU only version. Since the model is pretty small and I eventually want to run it with ort web purely with the wasm rt in browser, I didn't bother with the GPU version. The script I use to run the model is

```python
import onnxruntime
ort_session = onnxruntime.InferenceSession("det.onnx")
import numpy as np
ort_input = np.load("det_test_input.npy") 
ort_output  = ort_session.run(None, {'input': ort_input})
```

The result I got is

```
[array([[[[0., 0., 0., ..., 0., 0., 0.],
         [0., 0., 0., ..., 0., 0., 0.],
         [0., 0., 0., ..., 0., 0., 0.],
         ...,
         [0., 0., 0., ..., 0., 0., 0.],
         [0., 0., 0., ..., 0., 0., 0.],
         [0., 0., 0., ..., 0., 0., 0.]]]], dtype=float32)]
```

To get the result before sigmoid, I used `skl2onnx` to add all nodes into output.

```python
from skl2onnx.helpers.onnx_helper import enumerate_model_node_outputs
from skl2onnx.helpers.onnx_helper import load_onnx_model
from skl2onnx.helpers.onnx_helper import select_model_inputs_outputs
from skl2onnx.helpers.onnx_helper import save_onnx_model
onnx_model  = load_onnx_model("det.onnx")
all_nodes   = list(enumerate_model_node_outputs(onnx_model))
node_dict   = {name:idx for idx, name in enumerate(all_nodes)}
onnx_model_new = select_model_inputs_outputs(onnx_model, all_nodes)
save_onnx_model(onnx_model_new, "det_debug.onnx")
```

By running the inference again with `det_debug.onnx` and extract the output `ort_output[node_dict['705']]` according to the graph I see in netron (as shown in the screenshot at the beginning of the post). The result looks correct and is by no means extreme.

```
array([[[[-46.49633 , -47.637505, -48.94919 , ..., -42.49288 ,
          -44.376823, -45.41898 ],
         [-49.309967, -48.91755 , -51.42996 , ..., -44.173256,
          -46.379845, -47.28559 ],
         [-44.940323, -45.326736, -48.935936, ..., -41.55615 ,
          -44.305103, -45.24744 ],
         ...,
         [-45.143105, -44.42542 , -46.89444 , ..., -43.22701 ,
          -44.884735, -46.0697  ],
         [-41.901318, -42.161957, -45.74202 , ..., -39.71297 ,
          -44.009773, -44.926983],
         [-40.782604, -41.34737 , -42.222134, ..., -39.806213,
          -42.886993, -42.98987 ]]]], dtype=float32)
```

By manually performing sigmoid, you would get

```
array([[[[6.4106247e-21, 2.0478334e-21, 5.5161703e-22, ...,
          3.5121732e-19, 5.3381464e-20, 1.8827290e-20],
         [3.8455051e-22, 5.6934871e-22, 4.6158628e-23, ...,
          6.5433289e-20, 7.2026017e-21, 2.9115798e-21],
         [3.0385456e-20, 2.0646596e-20, 5.5897588e-22, ...,
          8.9617740e-19, 5.7350623e-20, 2.2350454e-20],
         ...,
         [2.4808403e-20, 5.0849387e-20, 4.3053022e-21, ...,
          1.6855768e-19, 3.2122344e-20, 9.8216453e-21],
         [6.3458381e-19, 4.8898420e-19, 1.3629919e-20, ...,
          5.6607762e-18, 7.7054556e-20, 3.0793511e-20],
         [1.9424062e-18, 1.1042446e-18, 4.6042657e-19, ...,
          5.1568106e-18, 2.3681840e-19, 2.1366702e-19]]]], dtype=float32)
```

I do noticed that if you used the naïve `np.exp` implementation of sigmoid, you will be encountered with a warning that says "RuntimeWarning: overflow encountered in exp". You can implement a numerically more stable sigmoid or use the `torch.sigmoid` after turning the numpy arry into pytorch tensor to avoid this issue, but I think if that is the problem, ort should be smart enough to handle all these within its implementation of sigmoid,

**Comment by xadupre** [comment]
I faced a similar issue and I compared onnxruntime to numpy implementation. onnxruntime gives very different results when `x < 1e16`. Is it a bug or is it expected? 

![image](https://user-images.githubusercontent.com/22452781/150138477-678c82e1-963a-4775-b5e8-f9343a7b5e5e.png)
